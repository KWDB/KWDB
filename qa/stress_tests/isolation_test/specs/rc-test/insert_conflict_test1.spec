setup
{
  CREATE TABLE ints (key int, val text, PRIMARY KEY (key));
}

teardown
{
  DROP TABLE ints;
}

session s1
step s1     { BEGIN transaction isolation level read committed; }
step s1w    { INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING; }
step s1c    { COMMIT; }
step show   { SELECT * FROM ints; }

session s2
step s2     { BEGIN transaction isolation level read committed; }
step s2w    { INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING; }
step s2c    { COMMIT; }

permutation s1 s2 s1w s1c s2w s2c show
permutation s1 s2 s2w s2c s1w s1c show
permutation s1 s2 s1w s2w s1c s2c show
permutation s1 s2 s2w s1w s2c s1c show
permutation s2 s1 s1w s1c s2w s2c show
permutation s2 s1 s2w s2c s1w s1c show
permutation s2 s1 s1w s2w s1c s2c show
permutation s2 s1 s2w s1w s2c s1c show
