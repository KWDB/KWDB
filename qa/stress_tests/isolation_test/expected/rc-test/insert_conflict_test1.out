Parsed test spec with 2 sessions

starting permutation: s1 s2 s1w s1c s2w s2c show
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s1c: COMMIT;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s2c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing1
(1 row)


starting permutation: s1 s2 s2w s2c s1w s1c show
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s2c: COMMIT;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s1c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing2
(1 row)


starting permutation: s1 s2 s1w s2w s1c s2c show
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING; <waiting ...>
step s1c: COMMIT;  finished.
step s2w: <... completed>
ERROR:  flow: memory budget exceeded: 10240 bytes requested, 0 currently allocated, 0 bytes in budget
step s2c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing1
(1 row)


starting permutation: s1 s2 s2w s1w s2c s1c show
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING; <waiting ...>
step s2c: COMMIT;  finished.
step s1w: <... completed>
ERROR:  flow: memory budget exceeded: 10240 bytes requested, 0 currently allocated, 0 bytes in budget
step s1c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing2
(1 row)


starting permutation: s2 s1 s1w s1c s2w s2c show
step s2: BEGIN transaction isolation level read committed;  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s1c: COMMIT;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s2c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing1
(1 row)


starting permutation: s2 s1 s2w s2c s1w s1c show
step s2: BEGIN transaction isolation level read committed;  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s2c: COMMIT;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s1c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing2
(1 row)


starting permutation: s2 s1 s1w s2w s1c s2c show
step s2: BEGIN transaction isolation level read committed;  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING; <waiting ...>
step s1c: COMMIT;  finished.
step s2w: <... completed>
ERROR:  flow: memory budget exceeded: 10240 bytes requested, 0 currently allocated, 0 bytes in budget
step s2c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing1
(1 row)


starting permutation: s2 s1 s2w s1w s2c s1c show
step s2: BEGIN transaction isolation level read committed;  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step s2w: INSERT INTO ints(key, val) VALUES(1, 'donothing2'), (1, 'donothing3') ON CONFLICT DO NOTHING;  finished.
step s1w: INSERT INTO ints(key, val) VALUES(1, 'donothing1') ON CONFLICT DO NOTHING; <waiting ...>
step s2c: COMMIT;  finished.
step s1w: <... completed>
ERROR:  flow: memory budget exceeded: 10240 bytes requested, 0 currently allocated, 0 bytes in budget
step s1c: COMMIT;  finished.
step show: SELECT * FROM ints;  finished.
key|val       
---+----------
  1|donothing2
(1 row)

